portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,8 +38,23 @@ public void init() {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    //TODO: modify so that doGet returns messages as a JSON string.
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello, Lily!</h1>");
+    String json = convertToJson(messages);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  private String convertToJson(List<String> messageList) {
+    String json = "{";
+    json += "\"MessageOne\": ";
+    json += "\"" + messages.get(0) + "\"";
+    json += ", ";
+    json += "\"MessageTwo\": ";
+    json += "\"" + messages.get(1) + "\"";
+    json += ", ";
+    json += "\"MessageThree\": ";
+    json += "\"" + messages.get(2) + "\"";
+    json += "}";
+    return json;
```

Comment:
Normally I would say that using hardcoded indexes here is a bad idea because if the size of messages changes, you will either get an out of bounds error or will be missing messages.

Reading the instructions though, it looks like they're just asking you con create a JSON object to pass down to the client that can then access the fields in the object you've created.  Since this is just for demo purposes though, I think this is okay.